### PR TITLE
TF 5298/5286 organize calc slug file and convert to ts

### PR DESCRIPTION
**Pull request by e-papanicolas**
### Ticket(s)

- _[[Airtable ticket 5298 - break file up](https://airtable.com/appfJZShN8K4tcWHU/pagxblGyhI5EJIA3v?HjEAY=rectLtzU1YnNpaJFf)]_
- _[[Airtable ticket 5286 - convert to ts](https://airtable.com/appfJZShN8K4tcWHU/pagxblGyhI5EJIA3v?HjEAY=recfTSOIhkS5L3LLV)]_

### Type of change

- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Description

This breaks the `[slug]` calculator template file into more manageable chunks which I'll refer to here as secondary components. Using intuitive break points, I sectioned the calculator off into individual non-export functions, which render inside the `calculatorSlugRoute` function as secondary components. The main component holds a few pieces of state or functionality, and the requirement to be in this main function is that they must be shared between more than one secondary component. Each of the secondary components holds individual state or functionality, and its JSX return. It is most easily viewed by minimizing each function so that you can see the bigger picture (as shown in the first image below). The main component is now much easier to understand and visualize (see the second image below). 

I updated the syntax on the final page conditionals so that it is easier to read and understand, and I was able to remove a few duplicate checks and condense items. 

I also went through each of the secondary components and cleaned up styling, eliminated any excess or duplicate, and added classnames so that debugging the entire file from the browser is also easier. 

During the conversion to typescript there were several errors that made themselves known, that javascript doesn't mind, so I made a small adjustment to the `portableTextComponent` and updated the call to print the PDF with the correct number of arguments. 

I typed the `getStaticProps` and `getStaticPaths` functions, and created interfaces to type all of the static props. I also created a `SharedCalcProps` interface to hold types for all the props that are NOT static and are passed around dynamically throughout the file. Each secondary component is then able to access all of these props as well.

### Screenshots

Include relevant screenshots.
Screenshot 2023-08-11 at 3 37 49 PM

Screenshot 2023-08-11 at 3 38 29 PM

### Checklist:

_Please delete any options that are not relevant_

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings in the console
- [x] There are no new linting errors/problems in my code
- [x] I have filled this PR out fully, and cleaned up any extraneous items so that it is clear and easy to understand

### CODE REVIEW

File: pages/calculator/[slug].tsx [hunk 1]
```diff
@@ -0,0 +1,597 @@
...
+      closeText: string;
+    };
+  };
+}
+
+export interface SharedCalcProps {
+  isFirstPage: () => boolean;
+  // eslint-disable-next-line no-unused-vars
+  addToResponses: (answer: string) => void;
+  setOpen: React.Dispatch<React.SetStateAction<boolean>>
```
Comment (mhyneman8) [inline pages/calculator/[slug].tsx:91]: I don't think you wrote this line, but let's add a more descriptive name here.

File: pages/calculator/[slug].tsx [hunk 2]
```diff
@@ -0,0 +1,597 @@
...
+  setShare: React.Dispatch<React.SetStateAction<boolean>>;
+  calcFirstPageUrl: string;
+  handleClose: () => void;
+  setShowResults: React.Dispatch<React.SetStateAction<boolean>>;
+  open: boolean;
```
Comment (mhyneman8) [inline pages/calculator/[slug].tsx:96]: Because this is also open, what is opening?

File: pages/calculator/[slug].tsx [hunk 3]
```diff
@@ -0,0 +1,597 @@
...
+  share: boolean;
```
Comment (mhyneman8) [inline pages/calculator/[slug].tsx:97]: And this one too. Is this just if the share button should show?

File: pages/calculator/[slug].tsx [hunk 4]
```diff
@@ -0,0 +1,597 @@
...
+function CalcHeader({ page, isFirstPage }:
+  { page: StaticCalcProps['page'],
+    isFirstPage: SharedCalcProps['isFirstPage']
+  }) {
+  const router = useRouter();
+
+  const isPageIncludedInStepper = () => {
+    const excludedPageSlug = 'head';
+    const isPartOfHead = page.slug.includes(excludedPageSlug); // exclude
+    const { isFinalPage } = page; // exclude
```
Comment (mhyneman8) [inline pages/calculator/[slug].tsx:109]: What are these // exclude comments?

File: pages/calculator/[slug].tsx [hunk 5]
```diff
@@ -0,0 +1,597 @@
...
+          id="back-button"
+          onClick={() => {
+            router.back();
+          }}
+          sx={{
+            marginLeft: 0,
+            fontWeight: 'normal',
+            fontSize: '14px',
+            color: 'primary',
+          }}
```
Comment (mhyneman8) [inline pages/calculator/[slug].tsx:128]: I think we can remove all of these

File: pages/calculator/[slug].tsx [hunk 6]
```diff
@@ -0,0 +1,597 @@
...
+          && page.choices.map((choice) => {
+            const linkTo = choice.linkTo
+              ? `/calculator/${choice.linkTo.slug.current}`
+              : '#';
+            const href = choice.isExternalLink ? choice.url : linkTo;
+            return (
+              <Button
+                key={choice._key}
+                variant="contained"
+                color="primary"
```
Comment (mhyneman8) [inline pages/calculator/[slug].tsx:180]: Can remove this color also

File: pages/calculator/[slug].tsx [hunk 7]
```diff
@@ -0,0 +1,597 @@
...
+      </Link>
+    </Box>
+  );
+}
+
+function FeedbackContainer({ page, calculatorConfig }: StaticCalcProps) {
+  return (
+    <Button
+      variant="contained"
+      color="primary"
```
Comment (mhyneman8) [inline pages/calculator/[slug].tsx:231]: Can delete this color too

File: pages/calculator/[slug].tsx [hunk 8]
```diff
@@ -0,0 +1,597 @@
...
+  justify?: boolean,
+}) {
+  return (
+    <Box
+      id="share-calc-container"
+      sx={{
+        display: 'flex',
+        gap: 1,
+        justifyContent: justify ? 'center' : 'flex-start',
+      }}
```
Comment (mhyneman8) [inline pages/calculator/[slug].tsx:256]: Can get rid of all these styles

File: pages/calculator/[slug].tsx [hunk 9]
```diff
@@ -0,0 +1,597 @@
...
+    <Box
+      id="finalpage-links-container"
+      sx={{
+        display: 'flex',
+        flexDirection: {
+          xs: 'column', md: 'row',
+        },
+        width: '100%',
+        justifyContent: 'space-evenly',
+        alignItems: 'center',
```
Comment (mhyneman8) [inline pages/calculator/[slug].tsx:292]: align-items not needed

File: pages/calculator/[slug].tsx [hunk 10]
```diff
@@ -0,0 +1,597 @@
...
+
+  const handleDownloadClick = () => {
+    // print section must be on the page before save as pdf will work
+    setShowResults(true);
+    setTimeout(() => { saveAsPDF(); }, 500);
+  };
+
+  return (
+    <Button
+      sx={{ display: 'block' }}
```
Comment (mhyneman8) [inline pages/calculator/[slug].tsx:341]: I don't see this style needed

**Comment by mhyneman8** [comment]
This is so much easier to understand! Great work. 
I'm wondering if we should break this out even further. Create a file for the props and then import. Same thing with the functional pieces. So the slug file only holds the pages (first page, question page, final page). And import everything those pages need. What do you think?

**Review by mhyneman8 — CHANGES_REQUESTED** [review]
Check Conversation tab

File: pages/calculator/[slug].tsx (hunk 9 above)
Comment (e-papanicolas) [reply pages/calculator/[slug].tsx:292]: On sm and xs the align items is needed.

File: pages/calculator/[slug].tsx (hunk 8 above)
Comment (e-papanicolas) [reply pages/calculator/[slug].tsx:256]: I removed the gap but the others are needed on xs and sm.

**Comment by e-papanicolas** [comment]
@mhyneman8 I created a new components folder `/components/calculator/misdemeanor` to hold all of the calculator pieces. I created a new file `/utils/calculator.props.ts` to hold both static and shared props.

**Review by mhyneman8 — APPROVED** [review]

**Pull request merged**